Specs/DateTime/English/DateTimeModel.json

```diff
@@ -21185,5 +21185,177 @@
+      "ReferenceDateTime": "2018-11-30T12:00:00"
+    },
+    "NotSupported": "java, javascript, python",
+    "Results": [
+      {
+        "Text": "morning",
+        "Start": 15,
+        "End": 21,
+        "TypeName": "datetimeV2.timerange",
+        "Resolution": {
+          "values": [
+            {
+              "timex": "TMO",
+              "type": "timerange",
+              "start": "08:00:00",
+              "end": "12:00:00"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "Input": "Let's meet one in the morning together",
+    "Context": {
+      "ReferenceDateTime": "2018-11-30T12:00:00"
+    },
+    "Results": [
+      {
+        "Text": "one in the morning",
+        "Start": 11,
+        "End": 28,
+        "TypeName": "datetimeV2.time",
+        "Resolution": {
+          "values": [
+            {
+              "timex": "T01",
+              "type": "time",
+              "value": "01:00:00"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "Input": "Let's go out 1 evening",
+    "Context": {
+      "ReferenceDateTime": "2018-11-30T12:00:00"
+    },
+    "NotSupported": "java, javascript, python",
+    "Results": [
+      {
+        "Text": "evening",
+        "Start": 15,
+        "End": 21,
+        "TypeName": "datetimeV2.timerange",
+        "Resolution": {
+          "values": [
+            {
+              "timex": "TEV",
+              "type": "timerange",
+              "start": "16:00:00",
+              "end": "20:00:00"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "Input": "Students must attend all four afternoon classes",
+    "Context": {
+      "ReferenceDateTime": "2018-11-30T12:00:00"
+    },
+    "NotSupported": "java, javascript, python",
+    "Results": [
+      {
+        "Text": "afternoon",
+        "Start": 30,
+        "End": 38,
+        "TypeName": "datetimeV2.timerange",
+        "Resolution": {
+          "values": [
+            {
+              "timex": "TAF",
+              "type": "timerange",
+              "start": "12:00:00",
+              "end": "16:00:00"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "Input": "I met them at 2 morning sessions",
+    "Context": {
+      "ReferenceDateTime": "2018-11-30T12:00:00"
+    },
+    "NotSupported": "java, javascript, python",
+    "Results": [
+      {
+        "Text": "morning",
+        "Start": 16,
+        "End": 22,
+        "TypeName": "datetimeV2.timerange",
+        "Resolution": {
+          "values": [
+            {
+              "timex": "TMO",
+              "type": "timerange",
+              "start": "08:00:00",
+              "end": "12:00:00"
+            }
+          ]
+        }
+      }
+    ]
+  },
+  {
+    "Input": "They work at three night markets.",
+    "Context": {
+      "ReferenceDateTime": "2018-11-30T12:00:00"
+    },
+    "Comment": "Currently not supported because 'three night' is extracted as duration.",
```

Comment:
Can you add a case for "night market/markets" in the ambiguity filter?

Reply:
yes, but there seem to be many other such expressions: "night shifts, night jobs, night clubs..."
And the same happens with other duration units, e.g.
"2 year long projects" -> "2 year",
"3 day hospitals" -> "3 day"...

Would not be better to address the issue in DurationExtractor? 
We already implemented a similar solution in DatePeriod and TimePeriod (#2634).

Reply:
Checking frequencies, at least "night market/markets, shift/shits, and club/clubs" should be in the ambiguity filter. 
There will of course still be cases not caught. But we can filter these frequent ones. If the error happens in duration, then the filter should be in that type.

Reply:
filter added